### PR TITLE
[AIRFLOW-1000] Rebrand distribution to Apache Airflow

### PR DESCRIPTION
**Pull request by bolkedebruin**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1000

### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:
Per Apache requirements Airflow should be branded Apache Airflow.
It is impossible to provide a forward compatible automatic update
path and users will be required to manually upgrade.

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
No tests, it is a semantic issue.

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

@aoen @criccomini @jlowin (BTW suggestion: Target for 1.9.0, as 1.8.X will be bug fix releases)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -99,6 +100,15 @@ def write_version(filename=os.path.join(*['airflow',
         a.write(text)
 
 
+def check_previous():
+    installed_packages = ([package.project_name for package
```
Comment (aoen) [inline setup.py:104]: Don't need the outer brackets

**Comment by aoen** [comment]
It's unfortunate we can't make the upgrade process more smooth but I know that you researched other methods/projects that made similar transitions so looks like this is the best we are going to get.

LGTM

**Comment by criccomini** [comment]
+1 LGTM

**Comment by bolkedebruin** [comment]
@criccomini needs to be cherry picked into 1.8

**Comment by bolkedebruin** [comment]
@criccomini done.

**Pull request closed without merging**